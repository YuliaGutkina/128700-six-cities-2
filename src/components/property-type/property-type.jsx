import PropTypes from "prop-types";


const TYPES = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

const PropertyType = (props) => {
  return TYPES[props.type];
};

PropertyType.propTypes = {
  type: PropTypes.oneOf(Object.keys(TYPES)).isRequired
};


export {PropertyType};
