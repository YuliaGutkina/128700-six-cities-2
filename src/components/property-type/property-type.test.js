import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import {PropertyType} from "./property-type";


it(`Favorites page correctly renders after relaunch`, () => {
  const renderer = new ShallowRenderer();
  renderer
    .render(
        <PropertyType type={`house`}/>);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

