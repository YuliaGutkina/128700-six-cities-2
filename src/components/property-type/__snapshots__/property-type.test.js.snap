// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites page correctly renders after relaunch 1`] = `"House"`;
